### PR TITLE
feat(session replay): add metadata to headers

### PR DESCRIPTION
**Pull request by jxiwang**
### Summary

Adding sample rate, full url, and version number to header. When adding metadata to the body, the request throws 400 error. Also added version file update script to match sdk-bot. 

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  Possible issues with SSR frameworks such as Next.js when accessing the window, but there is a check that should account for that in helper file under getCurrentUrl.

### CODE REVIEW

File: packages/session-replay-browser/src/constants.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export const DEFAULT_EVENT_PROPERTY_PREFIX = '[Amplitude]';
 export const DEFAULT_SESSION_REPLAY_PROPERTY = `${DEFAULT_EVENT_PROPERTY_PREFIX} Session Recorded`;
 export const DEFAULT_SESSION_START_EVENT = 'session_start';
 export const DEFAULT_SESSION_END_EVENT = 'session_end';
+export const DEFAULT_SAMPLE_RATE = 1;
```
Comment (kwalker3690) [inline packages/session-replay-browser/src/constants.ts:9]: Default sample rate should be 0

File: packages/session-replay-browser/src/session-replay.ts [hunk 2]
```diff
@@ -400,6 +406,9 @@ export class SessionReplay implements AmplitudeSessionReplay {
           'Content-Type': 'application/json',
           Accept: '*/*',
           Authorization: `Bearer ${apiKey}`,
+          'X-Client-Version': version,
```
Comment (kwalker3690) [inline packages/session-replay-browser/src/session-replay.ts:409]: Check with backend that they can parse these

File: packages/session-replay-browser/src/verstion.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export const VERSION = '0.2.5';
```
Comment (kwalker3690) [inline packages/session-replay-browser/src/verstion.ts:1]: Nit: file is misspelled

File: packages/session-replay-browser/test/session-replay.test.ts [hunk 4]
```diff
@@ -1007,6 +1016,15 @@ describe('SessionReplayPlugin', () => {
     });
   });
 
+  describe('getSampleRate', () => {
+    test('should return undefined if no config set', () => {
```
Comment (kwalker3690) [inline packages/session-replay-browser/test/session-replay.test.ts:1020]: should return default if no config set

File: packages/session-replay-browser/src/helpers.ts [hunk 5]
```diff
@@ -25,3 +25,8 @@ export const isSessionInSample = function (sessionId: number, sampleRate: number
   const mod = absHashMultiply % 100;
   return mod / 100 < sampleRate;
 };
+
+export const getCurrentUrl = () => {
+  // eslint-disable-next-line no-restricted-globals
+  return typeof window !== 'undefined' ? window.location.href : '';
```
Comment (kwalker3690) [inline packages/session-replay-browser/src/helpers.ts:31]: Can you use `getGlobalScope` here instead? You won't have to disable the lint rule then too and its a bit more standard in the codebase

Comment (jxiwang) [reply packages/session-replay-browser/src/helpers.ts:31]: Yeah I can. Did not realize there was a function for this already.

**Review by kwalker3690 — APPROVED** [review]

**Pull request merged**